src/content/blog/astro-470.mdx

```diff
@@ -0,0 +1,70 @@
+
+Full release highlights include:
+
+- [**Major improvements to the Dev Toolbar API**](#dev-toolbar-api-improvements)
+- [**Update checker**](#update-checker)
+- [**`allowJs: true` for strictest TypeScript preset**](#allowjs-true-for-strictest-typescript-preset)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Dev Toolbar API improvements
+
+Astro 4.7 includes major improvements to the API for making toolbar apps. One of our main goal this release was to make it easier to build and maintain toolbar apps, and we're excited to share these improvements with you!
+
+- `defineToolbarApp` helper that makes it easier to define toolbar apps, this is similar to the `defineConfig` helper for defining Astro config, or `defineMiddleware` for defining middleware.
+- `app` and `server` helpers to make it easier to interact with the toolbar and the server.
+- New starter project for building toolbar apps, to help you get started building your own toolbar apps in no time. Run `npm create astro@latest -- --template toolbar-app` and start building your toolbar app.
+- [Revamped documentation for building toolbar apps](https://docs.astro.build/en/reference/dev-toolbar-app-reference/).
+- [All new recipe guiding you through building a toolbar app from scratch](https://docs.astro.build/en/recipes/making-toolbar-apps/), including how to use JSX frameworks like React or Preact in your app.
+
+We hope you'll enjoy these improvements and we can't wait to see what you build with them!
+
+## Update checker
+
+Starting from this release, Astro will now check for updates when you run the dev server. If a new version is available, you'll see a message in the terminal with instructions on how to update and in the dev toolbar.
+
+<BlogContentImage src={updateCheckerImage} alt="A screenshot of the message that Astro shows when an update is available." />
+
+In order to avoid spamming you with update messages, Astro will only check for updates once per 10 days and only if you're multiple versions behind. You can also disable this feature by running `astro preferences disable checkUpdates` or setting the `ASTRO_DISABLE_UPDATE_CHECK` environment variable to `false`.
+
+## `allowJs: true` for strictest TypeScript preset
+
+Our `strictest` TypeScript preset previously included `allowJs: false`, to disable the usage of .js files completely. This proved to be a bit too strict for the average user and often caused confusion. We've now changed this to `allowJs: true`, which allows you to use .js files in your project.
+
+This should not be a breaking change, but if you're using the `strictest` preset and would like to still disable .js files, you can set `allowJs: false` manually in your `tsconfig.json`.
```

Comment:
```suggestion
Our `strictest` TypeScript preset previously included `allowJs: false`, to disable the use of `.js` files completely. This proved to be a bit too strict for the average user and often caused confusion. We've now changed this to `allowJs: true`, which allows you to use `.js` files in your project.

This should not be a breaking change, but if you're using the `strictest` preset and would like to still disable `.js` files, set `allowJs: false` manually in your `tsconfig.json`.
```
Mostly the suggestion here is that we just tend to use code syntax for file extensions or capitals for the name (e.g. PDF or `.pdf`) So either JS or `.js` would be consistent with that. I would keep it `.js` myself because we are specifically talking about files and their extensions, but your choice!